test/test_binary_ufuncs.py

```diff
@@ -1088,6 +1088,31 @@ def test_div_rounding_numpy(self, device, dtype):
                 actual, expect, exact_device=False, exact_dtype=exact_dtype
             )
 
+    @dtypes(*complex_types())
+    def test_complex_div_underflow_overflow(self, device, dtype):
+        # test to make sure the complex division does not produce underflow or overflow
+        # in the intermediate of its calculations
+        # NOTE: the calculation still produces an error if the number is greater than
+        # finfo.max / 2, but hopefully people realized that it's a dangerous region to work with
+        finfo = torch.finfo(dtype)
+        nom = torch.tensor([complex(finfo.min / 2, finfo.min / 2),
+                            complex(finfo.max / 2, finfo.max / 2),
```

Comment:
I think `(max + max j) / (max + max j)`  should give `1.0 + 0.0j`. Wrong result is due to overflow when directly computing `br + bi = max + max = 2 max` (same with `ar + ai`). Can we represent `x + x` as `((1/x) + (1/x)) ^ -1` to avoid overflow?

 
```
>>> import torch
>>> max = torch.finfo(torch.float32).max
>>> ar = torch.tensor(max)
>>> br = torch.tensor(max)
>>> bi = torch.tensor(max)
>>> ar = torch.tensor(max)
>>> ai = torch.tensor(max)
>>>
>>> rat = bi/br
>>> rat
tensor(1.)
>>>
>>>
>>> scl = 1.0 / (br + bi * rat)
>>> scl
tensor(0.) # due to br + bi overflow in the denom
>>> scl = (1/br) + (1/bi) * (1/rat)
>>> scl
tensor(5.8775e-39)
>>>
>>> num = (ar + ai * rat)
>>> num
tensor(inf) # overflow due to ar + ai 
>>> num = (1/ar) + (1/ai) * (1/rat)
>>> num
tensor(5.8775e-39)
>>>
>>> real_ = (1/num) * scl
>>> real_
tensor(1.)
```

Reply:
`a + b` is not equal to `(1/a + 1/b)^(-1)`.
Also, in my opinion we need to discuss whether overflow for number greater than `finfo.max / 2` worth making the computation slightly more expensive.

Reply:
I'd say it's fine if we overflow on such large numbers.